ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManagerMetrics.java

```diff
@@ -59,6 +60,11 @@ class HelixClusterManagerMetrics {
   public final Timer routingTableQueryTime;
   public final Counter resourceNameMismatchCount;
   public final Counter paranoidDurabilityIneligibleReplicaCount;
+  // These maps are updated when 'partitionCountWithOneReplicaDown' metric is populated. Rest of the metrics such as
+  // 'partitionCountWithTwoReplicaDown', 'partitionCountWithThreeReplicaDown' use these values instead of re-calculating
+  // the partitionCounts inorder to save cpu cycles.
+  public Map<Integer, Integer> localPartitionCountsByDownReplicas = new ConcurrentHashMap<>();
+  public Map<Integer, Integer> localPartitionCountsByNonLeaderStandbyReplicas = new ConcurrentHashMap<>();
```

Comment:
If these maps are not referenced outside of this metric cluster, we can make then privite.
Nit: an int[] would just be fine. The index would represent the number of down replicas and the value would represents how many partitions.

Reply:
Sure. Made them private. Since these might need be to be thread safe, for now, left them as ConcurrentHashMap instead of using AtomicInteger[].